storage/remote/queue_manager.go

```diff
@@ -332,6 +338,10 @@ func (t *QueueManager) Stop() {
 	for _, labels := range t.seriesLabels {
 		release(labels)
 	}
+	// Un-register metrics so we don't have alerts for queues that are gone.
+	for _, m := range t.queueMetrics {
+		prometheus.Unregister(m)
```

Comment:
Essentially, just repeat every `WithLabelaValues` call here, but use `Delete` instead of it.

Reply:
This works, except now if there's a config reload (which would call stop for all the queues) but the remote write config hasn't changed, the same queues come back up but there are no metrics for them. 

If the remote write config changes on each reload (adding a queue, removing a queue, etc.) then the metrics are always correct.

Reply:
I would expect that `metricVec.WithLabelValues` would panic here ^ when the metrics don't seem to exist.

Reply:
`metricVec.WithLabelValues` recreates the metric if it has been deleted before.
Whenever you take the `Delete` codepath, you have to make sure that the `WithLabelValues` codepath is taken, too, if the queue comes back up.

Reply:
That'd reset the metric value too, which is possibly not what we want.

Reply:
I found the cause of the problem. We in remote storage `ApplyConfig` we create new `QueueManagers`, then stop the old ones. When we stop the old ones we remove the metrics even though we might then be starting the exact same queue's in terms of config. Working on a fix right now.

Reply:
You should think about what @brian-brazil said: deleting and recreating the metrics will reset them, and you might not want that. I guess to avoid that, we need to track if a queue is created in the exact same way.

Reply:
BTW: You can always do the `WithLabelValues` call as it will simply return the existing metric if there is already one. If possible, just don't do the `DeleteLabelValues` if you know you'll recreate a queue with the same name.